src/app/index.js

```diff
@@ -27,11 +13,28 @@ var app = angular.module('mainApp', ['ngRoute'])
         // this.favorite_list.push(new Song('Jose Sabía', 'La Vela Puerca', 'De Bichos y Flores'));
         
         this.has_favorites = function(){
-            return this.favorite_list.length > 0;
-        }.bind(this);
+            return Favorite_Serv.favorite_list.length > 0;
+        };
 
-        this.add_favorite = function(title, artist, album){
+        this.changeView = function(route){
+            $location.path(route);
+        }
+
+    })
+    .service('Storage_Serv', function(){
```

Comment:
you should place this service in a separated file under /src/app/services/ folder